app/models/question.rb

```diff
@@ -43,6 +43,14 @@ def takes_placeholder?
     %w(text date).include? data_type
   end
 
+  def unique_prompt_name
```

Comment:
So, params[:responses] is still a hash when it gets sent to the controller, even if afterwards it is saved as an array. I could be completely off base with this, but I believe the duplicate keys will still get erased on the way to the controller.